tests/e2e/helpers/clusterpolicy.go

```diff
@@ -0,0 +1,131 @@
+	return h.client.NvidiaV1().ClusterPolicies().Update(ctx, cp, metav1.UpdateOptions{})
+}
+
+// modify applies a mutation function to a ClusterPolicy and persists the changes
+func (h *ClusterPolicyHelper) modify(ctx context.Context, name string, mutate func(*nvidiav1.ClusterPolicy)) error {
+	clusterPolicy, err := h.Get(ctx, name)
+	if err != nil {
+		return fmt.Errorf("failed to get ClusterPolicy: %w", err)
+	}
+
+	mutate(clusterPolicy)
+
+	_, err = h.Update(ctx, clusterPolicy)
+	if err != nil {
+		return fmt.Errorf("failed to update ClusterPolicy: %w", err)
+	}
+
+	return nil
+}
+
+func (h *ClusterPolicyHelper) UpdateDriverVersion(ctx context.Context, name, version string) error {
+	return h.modify(ctx, name, func(clusterPolicy *nvidiav1.ClusterPolicy) {
+		clusterPolicy.Spec.Driver.Version = version
+	})
+}
+
+func (h *ClusterPolicyHelper) EnableDCGM(ctx context.Context, name string) error {
+	return h.modify(ctx, name, func(clusterPolicy *nvidiav1.ClusterPolicy) {
+		clusterPolicy.Spec.DCGM.Enabled = ptr.To(true)
+	})
+}
+
+func (h *ClusterPolicyHelper) DisableDCGM(ctx context.Context, name string) error {
+	return h.modify(ctx, name, func(clusterPolicy *nvidiav1.ClusterPolicy) {
+		clusterPolicy.Spec.DCGM.Enabled = ptr.To(false)
+	})
+}
+
+func (h *ClusterPolicyHelper) EnableDCGMExporter(ctx context.Context, name string) error {
+	return h.modify(ctx, name, func(clusterPolicy *nvidiav1.ClusterPolicy) {
+		clusterPolicy.Spec.DCGMExporter.Enabled = ptr.To(true)
+	})
+}
+
+func (h *ClusterPolicyHelper) DisableDCGMExporter(ctx context.Context, name string) error {
+	return h.modify(ctx, name, func(clusterPolicy *nvidiav1.ClusterPolicy) {
+		clusterPolicy.Spec.DCGMExporter.Enabled = ptr.To(false)
+	})
+}
+
+func (h *ClusterPolicyHelper) EnableGFD(ctx context.Context, name string) error {
+	return h.modify(ctx, name, func(clusterPolicy *nvidiav1.ClusterPolicy) {
+		clusterPolicy.Spec.GPUFeatureDiscovery.Enabled = ptr.To(true)
+	})
+}
+
+func (h *ClusterPolicyHelper) DisableGFD(ctx context.Context, name string) error {
+	return h.modify(ctx, name, func(clusterPolicy *nvidiav1.ClusterPolicy) {
+		clusterPolicy.Spec.GPUFeatureDiscovery.Enabled = ptr.To(false)
+	})
+}
+
+func (h *ClusterPolicyHelper) SetMIGStrategy(ctx context.Context, name, strategy string) error {
+	return h.modify(ctx, name, func(clusterPolicy *nvidiav1.ClusterPolicy) {
+		clusterPolicy.Spec.MIG.Strategy = nvidiav1.MIGStrategy(strategy)
+	})
+}
+
+func (h *ClusterPolicyHelper) WaitForReady(ctx context.Context, name string, timeout time.Duration) error {
+	return wait.PollUntilContextTimeout(ctx, e2e.DefaultPollingInterval, timeout, true, func(ctx context.Context) (bool, error) {
+		clusterPolicy, err := h.Get(ctx, name)
+		if err != nil {
+			return false, err
+		}
+
+		for _, condition := range clusterPolicy.Status.Conditions {
+			if condition.Type == "Ready" && condition.Status == "True" {
```

Comment:
nit: use constants `internal/conditions.Ready` and `metav1.ConditionTrue` instead of hardcoding values.